tools/proguard5.3.2/test/proguard/util/ArrayUtilTest.java

```diff
@@ -0,0 +1,614 @@
+				new Comparable[]{1, 2}, 2,
+				new Comparable[]{1, 2, 3, 0, 0}, 5)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[0], 0,
+				new Comparable[0], 0)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2}, 3,
+				new Comparable[]{0, 1, 2, 3, 4}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2, 3, 4}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(1, ArrayUtil.compare(
+				new Comparable[]{2, 1, 0}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+	}
+
+	@Test
+	public void testExtendArrayBoolean() {
+		Assert.assertArrayEquals(new boolean[0],
+				ArrayUtil.extendArray(new boolean[0], 0)
+		);
+		Assert.assertArrayEquals(new boolean[]{true, true, true, false, false},
+				ArrayUtil.extendArray(new boolean[]{true, true, true}, 5)
+		);
+	}
+
+	@Test
+	public void testEnsureArraySizeBoolean() {
+		boolean[] booleans = new boolean[]{false, false, true};
+
+		Assert.assertArrayEquals(new boolean[]{true, true, true, true},
+				ArrayUtil.ensureArraySize(booleans, 4, true)
+		);
+
+		booleans = new boolean[]{false, false, true, true};
+
+		Assert.assertArrayEquals(new boolean[]{false, false, false, true},
+				ArrayUtil.ensureArraySize(booleans, 3, false)
+		);
+		Assert.assertArrayEquals(new boolean[]{false, false, false, false},
+				ArrayUtil.ensureArraySize(booleans, 4, false)
+		);
+	}
+
+	@Test
+	public void testAddByte() {
+		Assert.assertArrayEquals(new byte[]{0, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 0, (byte) 0)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 1, (byte) 0)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 0, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 3, (byte) 0)
+		);
+	}
+
+	@Test
+	public void testInsertByte() {
+		Assert.assertArrayEquals(new byte[]{1, 2, 3},
+				ArrayUtil.insert(new byte[]{1, 2}, 2, 2, (byte) 3)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 2, 2, 3},
+				ArrayUtil.insert(new byte[]{1, 2, 3}, 3, 2, (byte) 2)
+		);
+	}
+
+	@Test
+	public void testRemoveByte() {
+		byte[] bytes = new byte[]{1, 2, 3, 4};
+
+		ArrayUtil.remove(bytes, 4, 1);
+
+		Assert.assertArrayEquals(new byte[]{1, 3, 4, 0}, bytes);
+
+		bytes = new byte[]{1, 2, 3, 4};
+
+		ArrayUtil.remove(bytes, 4, 2);
+		ArrayUtil.remove(bytes, 3, 1);
+
+		Assert.assertArrayEquals(new byte[]{1, 4, 0, 0}, bytes);
+	}
+
+	@Test
+	public void testExtendArrayByte() {
+		Assert.assertArrayEquals(new byte[0],
+				ArrayUtil.extendArray(new byte[0], 0)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 2, 3, 0, 0},
+				ArrayUtil.extendArray(new byte[]{1, 2, 3}, 5)
+		);
+	}
+
+	@Test
+	public void testEnsureArraySizeByte() {
+		Assert.assertArrayEquals(new byte[]{2, 2, 2, 2},
+				ArrayUtil.ensureArraySize(new byte[]{1, 2, 3}, 4, (byte)2)
```

Comment:
add space between (byte) and 2.. same for the below code. Use the auto indent its super useful :)